src/frontends/onnx/frontend/src/op/com.microsoft/pad.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/com.microsoft/pad.hpp"
+
+#include "default_opset.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/op/pad.hpp"
+#include "ngraph/op/slice.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+OutputVector pad(const Node& node) {
+    auto inputs = node.get_ng_inputs();
+    auto data = inputs.at(0);
+    auto pads = inputs.at(1);
+
+    std::string pad_mode = node.get_attribute_value<std::string>("mode", "constant");
+    ov::op::PadMode mode;
+    ov::Output<ov::Node> pad_value;
+    if (pad_mode == "constant") {
+        mode = ov::op::PadMode::CONSTANT;
+        if (inputs.size() > 2) {
+            pad_value = inputs.at(2);
+        } else {
+            pad_value = default_opset::Constant::create(data.get_element_type(), Shape{}, {0});
+        }
+    } else if (pad_mode == "edge") {
+        mode = ov::op::PadMode::EDGE;
+    } else if (pad_mode == "reflect") {
+        mode = ov::op::PadMode::REFLECT;
+    } else {
+        throw ngraph_error("Unsupported pad_mode in ONNX com.microsoft.Pad operator");
+    }
+
+    auto pads_shape = pads.get_shape();
```

Comment:
The `.get_shape()` will throw for dynamic shape, it should be used only if static shape is ensured, 
`.get_partial_shape()` can be used instead. 
For partial shape `.is_static()` or `.is_dynamic()` can be used and `.size()` is safe only for shapes with static rank. 
To check whether the rank is static: `partial_shape.rank().is_static()`.